### ISSUE TITLE
libyaml changes broke master

### ISSUE THREAD

**Issue by ZeroChaos-** [hmatuschek/qdmr#141]
```
[213/333] : && /usr/bin/x86_64-pc-linux-gnu-g++ -Os -march=native -mtune=native -pipe -frecord-gcc-switches  -Wall -Wsign-compare -Wl,-O1 -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0    -rdynamic cli/CMakeFiles/dmrconf.dir/main.cc.o cli/CMakeFiles/dmrconf.dir/printprogress.cc.o cli/CMakeFiles/dmrconf.dir/detect.cc.o cli/CMakeFiles/dmrconf.dir/verify.cc.o cli/CMakeFiles/dmrconf.dir/readcodeplug.cc.o cli/CMakeFiles/dmrconf.dir/writecodeplug.cc.o cli/CMakeFiles/dmrconf.dir/encodecodeplug.cc.o cli/CMakeFiles/dmrconf.dir/decodecodeplug.cc.o cli/CMakeFiles/dmrconf.dir/infofile.cc.o cli/CMakeFiles/dmrconf.dir/writecallsigndb.cc.o cli/CMakeFiles/dmrconf.dir/encodecallsigndb.cc.o cli/CMakeFiles/dmrconf.dir/progressbar.cc.o -o cli/dmrconf -L/var/tmp/portage/net-wireless/qdmr-9999/work/qdmr-9999_build/src -Wl,-rpath,/var/tmp/portage/net-wireless/qdmr-9999/work/qdmr-9999_build/src:/var/tmp/portage/net-wireless/qdmr-9999/work/qdmr-9999_build/lib:  /usr/lib64/libQt5UiTools.a  -lusb-1.0  lib/libdmrconf.so.0.9.0  /usr/lib64/libQt5UiTools.a  /usr/lib64/libQt5Widgets.so.5.15.2  /usr/lib64/libQt5Gui.so.5.15.2  -lQt5Widgets  -lQt5Gui  -lQt5Core  -lQt5Widgets  -lQt5Gui  -lQt5Core  -lpthread  -lGL  /usr/lib64/libQt5Network.so.5.15.2  /usr/lib64/libQt5Positioning.so.5.15.2  /usr/lib64/libQt5SerialPort.so.5.15.2  /usr/lib64/libQt5Core.so.5.15.2  -lusb-1.0 && :
FAILED: cli/dmrconf 
: && /usr/bin/x86_64-pc-linux-gnu-g++ -Os -march=native -mtune=native -pipe -frecord-gcc-switches  -Wall -Wsign-compare -Wl,-O1 -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0    -rdynamic cli/CMakeFiles/dmrconf.dir/main.cc.o cli/CMakeFiles/dmrconf.dir/printprogress.cc.o cli/CMakeFiles/dmrconf.dir/detect.cc.o cli/CMakeFiles/dmrconf.dir/verify.cc.o cli/CMakeFiles/dmrconf.dir/readcodeplug.cc.o cli/CMakeFiles/dmrconf.dir/writecodeplug.cc.o cli/CMakeFiles/dmrconf.dir/encodecodeplug.cc.o cli/CMakeFiles/dmrconf.dir/decodecodeplug.cc.o cli/CMakeFiles/dmrconf.dir/infofile.cc.o cli/CMakeFiles/dmrconf.dir/writecallsigndb.cc.o cli/CMakeFiles/dmrconf.dir/encodecallsigndb.cc.o cli/CMakeFiles/dmrconf.dir/progressbar.cc.o -o cli/dmrconf -L/var/tmp/portage/net-wireless/qdmr-9999/work/qdmr-9999_build/src -Wl,-rpath,/var/tmp/portage/net-wireless/qdmr-9999/work/qdmr-9999_build/src:/var/tmp/portage/net-wireless/qdmr-9999/work/qdmr-9999_build/lib:  /usr/lib64/libQt5UiTools.a  -lusb-1.0  lib/libdmrconf.so.0.9.0  /usr/lib64/libQt5UiTools.a  /usr/lib64/libQt5Widgets.so.5.15.2  /usr/lib64/libQt5Gui.so.5.15.2  -lQt5Widgets  -lQt5Gui  -lQt5Core  -lQt5Widgets  -lQt5Gui  -lQt5Core  -lpthread  -lGL  /usr/lib64/libQt5Network.so.5.15.2  /usr/lib64/libQt5Positioning.so.5.15.2  /usr/lib64/libQt5SerialPort.so.5.15.2  /usr/lib64/libQt5Core.so.5.15.2  -lusb-1.0 && :
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `vtable for YAML::BadConversion'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::insert_map_pair(YAML::detail::node&, YAML::detail::node&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::memory::create_node()'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::set_style(YAML::EmitterStyle::value)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::set_null()'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `vtable for YAML::InvalidNode'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::set_tag(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `typeinfo for YAML::InvalidNode'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::set_scalar(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::end()'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::Emitter::~Emitter()'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::memory_holder::merge(YAML::detail::memory_holder&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::size() const'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::empty_scalar[abi:cxx11]()'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::set_type(YAML::NodeType::value)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::InvalidNode::~InvalidNode()'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `typeinfo for YAML::BadSubscript'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `typeinfo for YAML::BadConversion'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::convert_to_map(std::shared_ptr<YAML::detail::memory_holder> const&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::operator<<(YAML::Emitter&, YAML::Node const&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::begin()'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node::m_amount'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::BadConversion::~BadConversion()'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::push_back(YAML::detail::node&, std::shared_ptr<YAML::detail::memory_holder> const&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::LoadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::Emitter::c_str() const'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `vtable for YAML::RepresentationException'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `typeinfo for YAML::Exception'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::Emitter::Emitter()'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::BadSubscript::~BadSubscript()'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `vtable for YAML::Exception'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::detail::node_data::mark_defined()'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::Emitter::SetLocalValue(YAML::EMITTER_MANIP)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `YAML::convert<bool>::decode(YAML::Node const&, bool&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: lib/libdmrconf.so.0.9.0: undefined reference to `vtable for YAML::BadSubscript'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Comment by ZeroChaos-** [comment]
is it missing from the linker line or it is somehow just me?

**Comment by ZeroChaos-** [comment]
cmake is surprisingly quiet on the topic, but if it helps

```
>>> Configuring source in /var/tmp/portage/net-wireless/qdmr-0.9.0_alpha1/work/qdmr-0.9.0-alpha1 ...
 * Source directory (CMAKE_USE_DIR): "/var/tmp/portage/net-wireless/qdmr-0.9.0_alpha1/work/qdmr-0.9.0-alpha1"
 * Build directory  (BUILD_DIR):     "/var/tmp/portage/net-wireless/qdmr-0.9.0_alpha1/work/qdmr-0.9.0_alpha1_build"
cmake -C /var/tmp/portage/net-wireless/qdmr-0.9.0_alpha1/work/qdmr-0.9.0_alpha1_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Gentoo -DCMAKE_TOOLCHAIN_FILE=/var/tmp/portage/net-wireless/qdmr-0.9.0_alpha1/work/qdmr-0.9.0_alpha1_build/gentoo_toolchain.cmake  /var/tmp/portage/net-wireless/qdmr-0.9.0_alpha1/work/qdmr-0.9.0-alpha1
loading initial cache file /var/tmp/portage/net-wireless/qdmr-0.9.0_alpha1/work/qdmr-0.9.0_alpha1_build/gentoo_common_config.cmake
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/x86_64-pc-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/x86_64-pc-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found libusb-1.0:
--  - Includes: /usr/include/libusb-1.0
--  - Libraries: /usr/lib64/libusb-1.0.so
-- Build version 0.9.0
-- Using xsltproc: 
-- <<< Gentoo configuration >>>
Build type      Gentoo
Install path    /usr
Compiler flags:
C               -Os -march=native -mtune=native -pipe -frecord-gcc-switches
C++             -Os -march=native -mtune=native -pipe -frecord-gcc-switches  -Wall -Wsign-compare
Linker flags:
Executable      -Wl,-O1 -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0
Module          -Wl,-O1 -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0
Shared          -Wl,-O1 -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0

-- Configuring done
-- Generating done
-- Build files have been written to: /var/tmp/portage/net-wireless/qdmr-0.9.0_alpha1/work/qdmr-0.9.0_alpha1_build
```

**Comment by hmatuschek** [comment]
The yaml-cpp library is not imported by cmake as an external library but as a so-called target. The latter is possible if the library was itself build with cmake. This worked for me and should work if the config file for yaml-cpp is installed. For my system, it is under /usr/lib/x86_64-linux-gnu/cmake/yaml-cpp/yaml-cpp-config.cmake. I've now added a line that extracts the actual library path from the target and use that for linking.

**Comment by ZeroChaos-** [comment]
the following cmake files are installed with yaml-cpp in gentoo at this time.  If something is missing I can probably get it added pretty quickly/easily

```
/usr/lib/cmake/yaml-cpp/yaml-cpp-targets.cmake
/usr/lib/cmake/yaml-cpp/yaml-cpp-targets-gentoo.cmake
/usr/lib/cmake/yaml-cpp/yaml-cpp-config.cmake
/usr/lib/cmake/yaml-cpp/yaml-cpp-config-version.cmake
/usr/lib64/cmake/yaml-cpp/yaml-cpp-targets.cmake
/usr/lib64/cmake/yaml-cpp/yaml-cpp-targets-gentoo.cmake
/usr/lib64/cmake/yaml-cpp/yaml-cpp-config.cmake
/usr/lib64/cmake/yaml-cpp/yaml-cpp-config-version.cmake
```

**Comment by hmatuschek** [comment]
Fixed with release 0.9.0-alpha2.
